tommy susanto

TommySus opened the pull request:
success test 
CRUD + LOGIN

failed test 
CUD + LOGIN

SECRET=javascript
SALT=10

TommySus committed:
done

bayuoktari commented on _test_/product.test.js:
```diff
@@ -0,0 +1,411 @@
+
+//----------create error no token------------
+
+describe("error when Create Product", () => {
```
daripada buat describe berulang gini dibuat describe gede aja 
misalnya describe gedenya `POST /product`, di dalamnya ada describe `Success Create Product` & describe `Error When Create Product`

TommySus committed:
revisi-login

TommySus committed:
add-getProductById

TommySus committed:
add-env-template

TommySus committed:
done

TommySus committed:
done

TommySus committed:
add-api-doc

bayuoktari commented on _test_/user.test.js:
```diff
@@ -0,0 +1,106 @@
+            .post("/login")
+            .send({
+                user: "",
+                password: ""
```
user ???????

TommySus committed:
add server feature for customer
1. add cart controller
2. add history controller
3. add cart route
4. add history route
5. add cart fetch,create,delete,update
6. add history create

TommySus committed:
add include product on cart & history controller

TommySus committed:
revisi user controller

TommySus committed:
done

TommySus committed:
done

TommySus committed:
done

TommySus committed:
done

TommySus committed:
done

TommySus committed:
revisi-history

TommySus committed:
done
